### PR TITLE
19 - Paging

### PR DESCRIPTION
**Pull request by mar7in91Wolox**
# Summary
- I added Paging by a drop-down list.

# Trello Card
https://trello.com/c/yCMweZsh

# Screenshots
image
image
image

### CODE REVIEW

File: training-net/src/Controllers/MovieController.cs [hunk 1]
```diff
@@ -38,7 +44,8 @@ public IActionResult Index(string movieGenre, string searchString, string sortOr
                 movies = movies.Where(m => m.Genre == movieGenre);
             var moviesAndGenresVM = new MoviesAndGenresViewModel();
             moviesAndGenresVM.GenreList = new SelectList(genreQuery);
-            moviesAndGenresVM.MovieList = movies.Select(
+
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: Remove this empty line

**Comment by mar7in91Wolox** [comment]
Here was the problem:
image

File: training-net/src/Controllers/MovieController.cs [hunk 2]
```diff
@@ -37,55 +41,58 @@ public IActionResult Index(string movieGenre, string searchString, string sortOr
...
-                Genre = movie.Genre,
-                Price = movie.Price,
-                Rating = movie.Rating
-            }
-            ).ToList();           
             switch (sortOrder)
                 {
                     case "rating_desc":
-                        moviesAndGenresVM.MovieList = moviesAndGenresVM.MovieList.OrderByDescending(m => m.Rating).ToList();
+
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: Remove this empty line

File: training-net/src/Controllers/MovieController.cs [hunk 3]
```diff
@@ -37,55 +39,58 @@ public IActionResult Index(string movieGenre, string searchString, string sortOr
...
             switch (sortOrder)
                 {
                     case "rating_desc":
-                        moviesAndGenresVM.MovieList = moviesAndGenresVM.MovieList.OrderByDescending(m => m.Rating).ToList();
+                    
+                        movies = movies.OrderByDescending(m => m.Rating).ToList();
                         break;
                     case "rating":
-                        moviesAndGenresVM.MovieList = moviesAndGenresVM.MovieList.OrderBy(m => m.Rating).ToList();
+                       movies =movies.OrderBy(m => m.Rating).ToList();
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: Don't forget to space this statements correctly
` movies = movies.OrderBy(...);`

File: training-net/src/Controllers/MovieController.cs [hunk 4]
```diff
@@ -37,55 +39,58 @@ public IActionResult Index(string movieGenre, string searchString, string sortOr
...
+            movie => new MovieViewModel
+            {
+                Id = movie.Id,
+                Title = movie.Title,
+                ReleaseDate = movie.ReleaseDate,
+                Genre = movie.Genre,
+                Price = movie.Price,
+                Rating = movie.Rating
+            }
+            ).ToList();
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: move this line up
`}).ToList();`

File: training-net/src/Models/PaginatedList.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public class PaginatedList<T> : List<T>
+    {
+        public int PageIndex { get; private set; }
+        public int TotalPages { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int pageIndex, int pageSize)
+        {
+            PageIndex = pageIndex;
+            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+
```
Comment (truccomarcos) [inline training-net/src/Models/PaginatedList.cs]: Remove the empty line here

File: training-net/src/Models/PaginatedList.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            this.AddRange(items);
+        }
+
+        public bool HasPreviousPage
+        {
+            get
```
Comment (truccomarcos) [inline training-net/src/Models/PaginatedList.cs]: this get could be resolved in one line
`get { return (pageIndex > 1); }`

Same goes for the other ones

File: training-net/src/Controllers/MovieController.cs (hunk 4 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Done

File: training-net/src/Controllers/MovieController.cs (hunk 2 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Done

File: training-net/src/Controllers/MovieController.cs (hunk 3 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Done!

File: training-net/src/Models/PaginatedList.cs (hunk 6 above)
Comment (mar7in91Wolox) [reply training-net/src/Models/PaginatedList.cs]: Done

**Pull request merged**